### PR TITLE
Lib staking bindings

### PR DESCRIPTION
**Pull request by cryptoAtwill**
Expose `LibStaking` and `LibStakingChangeLog` bindings so that `ipc-provider` can query the `NewStakingRequest` event.

### CODE REVIEW

File: src/lib/LibStakingChangeLog.sol [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.19;
```
Comment (adlrocha) [inline src/lib/LibStakingChangeLog.sol:2]: Out of a curiosity, why did you decide to move this into its own library? I am fine with this, it is cleaner, but would love to understand the reasoning behind it :)

**Review by adlrocha — APPROVED** [review]
Good catch, lgtm. Thanks!

File: src/lib/LibStakingChangeLog.sol (hunk 1 above)
Comment (cryptoAtwill) [reply src/lib/LibStakingChangeLog.sol:2]: It's because the `build.rs` requires it to be in a separate file.

Comment (cryptoAtwill) [reply src/lib/LibStakingChangeLog.sol:2]: no choice if we dont change build.rs and it's quite simple to change.

**Pull request merged**